### ISSUE TITLE
DB store : Audio metadata

### ISSUE THREAD

**Issue by SurfaceS** [UniversalMediaServer/UniversalMediaServer#4742]
Just a reminder for me that audio db should have a cleanup to reflect other implementations.

An ALBUM is quite similar to TVSERIES.
Not media linked (sometimes a playlist).
It should have its own table MediaTableAudioAlbums -> AUDIO_ALBUMS.
MUSIC_BRAINZ_RELEASE_LIKE should be deleted (use AUDIO_ALBUMS for this).
MUSIC_BRAINZ_RELEASES should be deleted (use AUDIO_ALBUMS for this, like TMDB etc).

AUDIO_METADATA -> ~~song~~ media (it is an audio media, not always a song) itself, linked to a fileid.
Here it can link to an album (or not, like VIDEO_METADATA <-> TV_SERIES).
Remove all related things to ALBUM, use ALBUMID from AUDIO_ALBUMS.
Refactor COL_SONGNAME for COL_TITLE.

**Comment by SurfaceS** [comment]
@ik666 

> I can have a look at the MusicBrainzAPI, however, I didn't write it.

I know, but I think me and @SubJunk are not using UMS for audio, so I think we need you to have a look and confirm I do not break something on this side :)

> I think there is no real "album" bracket in the database yet

I'll back during the week with a massive update about AUDIO db and things.

> From a workflow perspective: If a user adds a MusicBrainzID to his music track(s) with let's say with "Piccard" or songkong, usually the other fields (like author, date, etc.) are also set correctly. So there is no real need to collect them over the MusicBrainz API again

We should parse the file for metadata, then we can use MusicBrainz on a background thread to fill fields that are not set (like we do for video).

> I added the my albums = liked albums and MUSIC_BRAINZ_RELEASE_LIKE table, because navigating the CDS by media library-> album is somehow cumbersome. If you have to many albums, you lose the overview.

On the massive update, it will be a MUSIC_ALBUM table.

I'll also split the "LIKE" things, as it is a user field, that should sit under it's own table (multi-users = multi-rating).
Also, like and rating are the same. One is true/false, the other is a range.
I think it is better to keep only the rating, ranged from 0.0 to 5.0

Like for movie, the CDS can browse by rating, then you don't loose the overview if you have too many.
But this is another story, we should first adapt the DB store.

> Maybe the MusicBrainzReleases table can be deleted. Looks like all columns already exists in AUDIO_METADATA (except the rating which can be miggtrated to MUSIC_BRAINZ_RELEASE_LIKE).

AUDIO_METADATA is about an audio media, can be an album track, but also other things (radio for ex, or book reading).

About MusicBrainzReleases, it seems that it store wrongly the song name for title (can you confirm as I don't have a lot music for testing ?). Should be the album title.
Those data will sit under the created MUSIC_ALBUM table.

**Comment by ik666** [comment]
@SurfaceS  Yes, I'll look from an audio perspective on those points. But first, you do the DB changes, right?

**Comment by SurfaceS** [comment]
yes

**Comment by ik666** [comment]
One more idea: It would actually be nice, to rate any object (playlist, folder, person, ...). Maybe we can add a rating column to the `StoreIds` table. Keeping an "overview" over persons/composer/albums folder with many entries can then be done as suggested by rating.

**Comment by SurfaceS** [comment]
`StoreIds` is not user aware.
What I can do for now is FileId rating (audio, video and so on) / TVSeries rating and Music Album rating.


> About MusicBrainzReleases, it seems that it store wrongly the song name for title (can you confirm as I don't have a lot music for testing ?).
can you confirm, then I can create the DB updater.

**Comment by ik666** [comment]
> > About MusicBrainzReleases, it seems that it store wrongly the song name for title (can you confirm as I don't have a lot music for testing ?).
> 
> can you confirm, then I can create the DB updater.

I've checked it. The `MUSIC_BRAINZ_RELEASES.TITLE` column has actually the album title and is therefore correct.

**Comment by SubJunk** [comment]
I confirm I don't use UMS for audio. I can always test audio things though
